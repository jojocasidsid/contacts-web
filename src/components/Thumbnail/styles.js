import styled from '@emotion/styled';
import PersonIcon from '@mui/icons-material/Person';

export const ThumbnailWrapper = styled('div')`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px;
  flex-direction: column;
  border: 1px solid ${({ theme }) => theme.palette.primary.main};
`;

export const ActionWrapper = styled('div')`
  & > button {
    margin: 20px;
  }
`;

export const StyledImage = styled(PersonIcon)`
  font-size: 200px;
`;
