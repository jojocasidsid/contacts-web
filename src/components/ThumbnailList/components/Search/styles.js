import TextField from '@mui/material/TextField';
import styled from '@emotion/styled';

export const StyledTextField = styled(TextField)`
  .MuiInput-root {
    border: 1px solid #30364333;
    border-radius: 2px;
    padding-left: 16px;
    min-width: 230px;
  }

  .MuiInput-root input {
    &::placeholder {
      font-weight: 600;
      font-size: 12px;
      font-style: normal;
    }
  }

  .MuiInputBase-root {
    margin-top: 23px;
  }

  .MuiInputLabel-root {
    color: #303643;
    opacity: 0.5;
    font-size: 18px;
    font-weight: bold;
  }

  .MuiInputLabel-root .Mui-focused {
    color: #303643;
  }
`;
