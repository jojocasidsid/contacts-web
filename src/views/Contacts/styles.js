import styled from '@emotion/styled';

export const ContactsWrapper = styled('div')``;
