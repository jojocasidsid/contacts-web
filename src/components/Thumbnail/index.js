import { Button, Typography } from '@mui/material';
import React from 'react';
import { ThumbnailWrapper, ActionWrapper, StyledImage } from './styles';

const Thumbnail = ({ data, handleEdit, handleDelete }) => (
  <ThumbnailWrapper>
    <div>
      <StyledImage color="primary" />
    </div>

    <Typography variant="body">Name: {data.name}</Typography>
    <Typography variant="body">Email: {data.email}</Typography>
    <Typography variant="body">Phone: {data.phone}</Typography>

    <ActionWrapper>
      <Button variant="contained" onClick={() => handleEdit(data.id)}>
        Edit
      </Button>
      <Button variant="outlined" onClick={() => handleDelete(data.id)}>
        Delete
      </Button>
    </ActionWrapper>
  </ThumbnailWrapper>
);

export default Thumbnail;
