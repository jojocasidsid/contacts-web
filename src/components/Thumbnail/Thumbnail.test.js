import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/material/styles';

import theme from 'config/theme/theme';

import Thumbnail from '.';

describe('Thumbnail', () => {
  it('should render without crashing', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Thumbnail
          data={{ name: 'jojo', id: 1, email: 'jvcasdisd20231@gmail.com', phone: '09057680922' }}
          handleEdit={() => {}}
          handleDelete={() => {}}
        />
      </ThemeProvider>
    );

    expect(tree).toMatchSnapshot();
  });
});
