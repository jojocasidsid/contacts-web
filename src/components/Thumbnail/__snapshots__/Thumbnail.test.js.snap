// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail should render without crashing 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #002F5D;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: #002F5D;
  font-size: 200px;
}

.emotion-2 {
  margin: 0;
  color: #282B2F;
  font-style: normal;
  font-weight: 300;
  font-size: 16px;
  line-height: 24px;
}

.emotion-5>button {
  margin: 20px;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  color: #fff;
  font-style: normal;
  font-weight: 600;
  font-size: 16x;
  line-height: 22px;
  letter-spacing: -0.02em;
  font-family: Rubik;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #002F5D;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.emotion-6::-moz-focus-inner {
  border-style: none;
}

.emotion-6.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-6 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-6:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(0, 32, 65);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-6:hover {
    background-color: #002F5D;
  }
}

.emotion-6:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-6.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-6.Mui-disabled {
  color: #282B2F;
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  color: #002F5D;
  font-style: normal;
  font-weight: 600;
  font-size: 16x;
  line-height: 22px;
  letter-spacing: -0.02em;
  font-family: Rubik;
  text-transform: uppercase;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(0, 47, 93, 0.5);
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 47, 93, 0.04);
  border: 1px solid #002F5D;
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-disabled {
  color: #282B2F;
  border: 1px solid rgba(0, 0, 0, 0.12);
}

<div
  className="emotion-0"
>
  <div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium emotion-1"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <span
    className="MuiTypography-root MuiTypography-body emotion-2"
  >
    Name: 
    jojo
  </span>
  <span
    className="MuiTypography-root MuiTypography-body emotion-2"
  >
    Email: 
    jvcasdisd20231@gmail.com
  </span>
  <span
    className="MuiTypography-root MuiTypography-body emotion-2"
  >
    Phone: 
    09057680922
  </span>
  <div
    className="emotion-5"
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  emotion-6"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Edit
    </button>
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  emotion-7"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
